sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.associateMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,556 @@
+        applicationStatus: ApplicationStatus.Completed,
+        firstDisbursementInitialValues: {
+          disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+        },
+        offeringInitialValues: {
+          studyStartDate: getISODateOnlyString(addDays(-1000)),
+          studyEndDate: getISODateOnlyString(addDays(-731)),
+        },
+        createSecondDisbursement: true,
+      },
+    );
+
+    // Create student and save fake application with disbursements.
+    const application = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+        createSecondDisbursement: true,
+      },
+    );
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+    // Fetch MSFAA Number for the student in SIMS.
+    const createdMSFAANumber = await db.msfaaNumber.findOne({
+      select: {
+        id: true,
+        msfaaNumber: true,
+        dateSigned: true,
+        dateRequested: true,
+        serviceProviderReceivedDate: true,
+        referenceApplication: { id: true },
+      },
+      relations: {
+        referenceApplication: true,
+      },
+      where: {
+        student: { id: student.id },
+      },
+      order: {
+        id: { direction: "DESC" },
+      },
+    });
+    // Assert MSFAA Number.
+    expect(createdMSFAANumber.msfaaNumber).not.toBe(null);
+    expect(createdMSFAANumber.msfaaNumber).not.toBe(
+      savedMSFAANumber.msfaaNumber,
+    );
+    // Assert date signed.
+    expect(createdMSFAANumber.dateSigned).toBe(null);
+    // Assert date requested.
+    expect(createdMSFAANumber.dateRequested).toBe(null);
+    // Assert service provider date.
+    expect(createdMSFAANumber.serviceProviderReceivedDate).toBe(null);
+  });
+
+  it("Should throw Disbursement not found exception when MSFAA Number is tried to be created before disbursement.", async () => {
+    // Arrange
+    // Create save fake application without disbursements.
+    const application = await saveFakeApplication(db.dataSource);
+
+    // Act
+    const associateMSFAAPayload = createFakeAssociateMSFAAPayload({
+      assessmentId: application.currentAssessment.id,
+    });
+    const saveResult = await disbursementController.associateMSFAA(
+      associateMSFAAPayload,
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(saveResult)).toBe(
+      MockedZeebeJobResult.Error,
+    );
+  });
+
+  it("Should throw MSFAA number is already associated when MSFAA Number is already associated to a disbursement.", async () => {
```

Comment:
`throw` can be misleading because the method is still returning `Complete`.
Suggestion:
```
Should return 'Completed' when the MSFAA is already associated.
```